android-sdk/src/main/java/com/mobilecoin/lib/OwnedTxOut.java

```diff
@@ -161,8 +160,8 @@ public RistrettoPublic getTargetKey() {
     }
 
     @NonNull
-    public RistrettoPublic getSharedSecret() {
-        return sharedSecret;
+    public RistrettoPublic getSharedSecret(AccountKey accountKey) throws TransactionBuilderException {
```

Comment:
@stevenspiel Will this be OK for you?

Reply:
Overall performance is better this way because you calculate `shared_secret` only when you need it instead of upon the creation of each `OwnedTxOut` instance. If this method is intended to be called multiple times we can keep the `sharedSecret` as a class field and return it upon subsequent queries, just skip that field when the object is serialized.